src/cli.h

```diff
@@ -4,39 +4,28 @@
 #include <stdint.h>
 
 #include "logging.h"
-#include "errcodes.h"
 
+/**
+ * @brief Stores passed cli arguments for later use
+ */
 typedef struct CLIArguments {
+	/**
+	 * @brief Indicates the minimum logging level (checked by ::log_str())
+	 */
     LoggingLevel logging_level;
+	/**
+	 * @brief Propagated to ::rom_load() to create a memory map to rom file
+	 */
     const char* rom_path;
 } CLIArguments;
 
-/**
- * @brief   Allocates a struct to store CLI arguments in and returns a pointer to that struct
- *
- * @warning The allocated memory must be freed by using @p cli_config_destroy 
- * @returns Pointer to a CLIArguments struct
- */
-CLIArguments* cli_config_default(void);
-
-/**
- * @brief   Deallocates the memory, where @p conf points to
- *
- * @param   conf    Pointer to @p CLIArguments
- */
-void cli_config_destroy(CLIArguments *const conf);
-
 /**
  * @brief   Extract CLI flags with (optional) arguments from @p argv
  *
- * @param   conf    Pointer to @p CLIArguments
- * @param   argc    Number of arguments passed to the program
- * @param   argv    The passed arguments
- *
- * @returns An error code
- * @retval  ERR_SUCCESS on success
- * @retval  ERR_FAILURE otherwise
+ * @param   cli_args	Will be filled with parsed values from CLI
+ * @param   argc    	Number of arguments passed to the program
+ * @param   argv    	The passed arguments
```

Comment:
I think noting down, that this function might not return could be useful.

Reply:
"this function" being `cli_parse`

Reply:
That's a good idea. I think we should generally do that for functions that are exposed but may exit.